foundry_scripts/InjectorInfraMultiChainDeployment.s.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.25;
+
+import {Script} from "forge-std/Script.sol";
+
+import {ChildChainGaugeInjectorV2} from "../contracts/ChildChainGaugeInjectorV2.sol";
+import {ChildChainGaugeInjectorV2Factory} from "../contracts/injectorFactoryV2.sol";
+
+/// @notice Deploys the v2 infrastructure for the injectors in all chains in `foundry.toml` in the following order:
+/// 1. {ChildChainGaugeInjectorV2} -> singleton/implementation purposes (helps verifying in etherscan etc)
+/// 2. {ChildChainGaugeInjectorV2Factory}
+contract InjectorInfraMultiChainDeployment is Script {
+    enum Chains {
+        ARBITRUM,
+        BASE,
+        ETHEREUM,
+        FRAXTAL,
+        GNOSIS,
+        OPTIMISM,
+        POLYGON
+    }
+
+    // injector infrastructure
+    ChildChainGaugeInjectorV2 injectorImpl;
+    ChildChainGaugeInjectorV2Factory injectorFactory;
+
+    mapping(Chains chain => string rpcAlias) public availableChains;
+
+    constructor() {
+        availableChains[Chains.ARBITRUM] = "arbitrum";
+        availableChains[Chains.BASE] = "base";
+        availableChains[Chains.ETHEREUM] = "ethereum";
+        availableChains[Chains.FRAXTAL] = "fraxtal";
+        availableChains[Chains.GNOSIS] = "gnosis";
+        availableChains[Chains.OPTIMISM] = "optimism";
+        availableChains[Chains.POLYGON] = "polygon";
+    }
+
+    /// @dev broadcast transaction modifier
+    /// @param pk private key to broadcast transaction
+    modifier broadcast(uint256 pk) {
+        vm.startBroadcast(pk);
+
+        _;
+
+        vm.stopBroadcast();
+    }
+
+    function run() public {
+        // read pk from `.env`
+        uint256 pk = vm.envUint("PRIVATE_KEY");
+
+        // @note the array can be updated depending on your target chains to deploy
+        // @note by default the script will deploy in all chains available in the toml file
+        Chains[] memory targetDeploymentChains = new Chains[](6);
```

Comment:
here you can target the specifics chains which you will like to deploy at once, remove as you may see fit, but also remember to adapt the length of the array